senderworker/src/main/scala/com/gu/notifications/ec2worker/SenderWorker.scala

```diff
@@ -4,57 +4,91 @@ import com.amazonaws.services.lambda.runtime.events.SQSEvent.SQSMessage
 
-import scala.concurrent.Future
+import java.util.concurrent.TimeUnit
+import javax.jms
+import javax.jms.{Destination, MessageConsumer, MessageListener, ObjectMessage, Queue, Session, TextMessage}
+import javax.security.auth.callback.Callback
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
 
 object SenderWorker extends App {
 
   implicit val logger: Logger = LoggerFactory.getLogger(this.getClass)
 
-  val sqsClient = AmazonSQSAsyncClientBuilder.standard()
+  val sqsClient = AmazonSQSClientBuilder.standard()
     .withCredentials(credentialsProvider)
     .withRegion("eu-west-1")
     .build()
 
-  def pollQueue(queue: String, appName: String, handleChunkTokens: (SQSEvent, Context) => Unit): Future[Unit] = Future {
-    logger.info(s"About to poll queue - $appName")
-    val receiveMessages = sqsClient.receiveMessage(new ReceiveMessageRequest().withQueueUrl(queue).withWaitTimeSeconds(20)).getMessages.asScala.toList
-
-    def parsedReceivedMessages(message: List[Message]): SQSEvent = {
-      val parsedMessages = message.map(m => {
-        val sqsMessage = new SQSMessage()
-        sqsMessage.setBody(m.getBody)
-        sqsMessage.setAttributes(m.getAttributes)
-        sqsMessage.setReceiptHandle(m.getReceiptHandle)
-        sqsMessage
-      })
-
-      val event = new SQSEvent()
-      event.setRecords(parsedMessages.asJava)
-      event
+  class OpenConnection(queueName: String) {
+    val connectionFactory = new SQSConnectionFactory(
+      new ProviderConfiguration(),
+      sqsClient
+    )
+
+    val connection: SQSConnection = connectionFactory.createConnection()
+
+    val connectionSession: Session = connection.createSession(false, Session.CLIENT_ACKNOWLEDGE)
+
+    val consumer: MessageConsumer = connectionSession.createConsumer(connectionSession.createQueue(queueName))
+
+    connection.start()
+  }
+
+  class ReceiverCallback(queueUrl: String, handleChunkTokens: (SQSEvent, Context) => Unit) extends MessageListener {
+
+    def parsedReceivedMessages(message: jms.Message): Option[SQSEvent] = {
+
+     message match {
+        case textMessage: TextMessage => {
+          val sqsMessage = new SQSMessage()
+          sqsMessage.setBody(textMessage.getText)
+          sqsMessage.setAttributes(Map("SentTimestamp" -> textMessage.getJMSTimestamp.toString).asJava)
+          val event = new SQSEvent()
+          event.setRecords(List(sqsMessage).asJava)
+          Some(event)
+          }
+        case _ => None
+      }
     }
 
-    val parsedEvent: SQSEvent = parsedReceivedMessages(receiveMessages)
+    override def onMessage(message: jms.Message): Unit = {
+      println("JMS message received")
+      val parsedEvent = parsedReceivedMessages(message)
+      parsedEvent match {
+        case Some(event) => {
+          handleChunkTokens(event, null)
+          message.acknowledge()
+        }
+        case None => {
+          logger.error("Cannot parse message")
```

Comment:
We tried to test what happens if we reached this branch, but we couldn't (why testing locally). For now, we think what'll happen is:
- we'll log a message
- no error will be thrown
- no message.acknowledge --> message remains on queue (?)

We thought that maybe this would be thought about a little more at a later stage